src/cryptojwt/key_bundle.py

```diff
@@ -395,10 +395,12 @@ def do_remote(self):
             if hasattr(_http_resp, "headers"):
                 headers = getattr(_http_resp, "headers")
                 self.last_remote = headers.get("last-modified") or headers.get("date")
+            res = True
 
         elif _http_resp.status_code == 304:  # Not modified
             LOGGER.debug("%s not modified since %s", self.source, self.last_remote)
             self.time_out = time.time() + self.cache_time
+            res = False
```

Comment:
I don't like these assignments `res = True` `res = False` - they don't speak of anything - and then there's a nothing-done-with-res special `else` case. This should be rewritten to express the intent clearly. For a start I would propose the  following, that makes the states explicit:

```python
...

load_successful = _http_resp.status_code == 200
not_modified = _http_resp.status_code == 304
error_response = not load_successful and not not_modified

if error_reponse:
    ... raise ...

if not_modified:
    ...

if load_successful:
    ...

self.last_updated = time.time()
return load_successful
```

Reply:
Agree, but I think using elif and an else to handle the error case is more inline with how the rest of the code is written. Pushed updated version.